rpc/jsonrpc/server/rpc_func.go

```diff
@@ -225,3 +240,8 @@ func isNullOrEmpty(params json.RawMessage) bool {
 		bytes.Equal(params, []byte("{}")) ||
 		bytes.Equal(params, []byte("[]"))
 }
+
+// isByteArray reports whether t is (equivalent to) []byte.
+func isByteArray(t reflect.Type) bool {
+	return t.Kind() == reflect.Slice && t.Elem().Kind() == reflect.Uint8
```

Comment:
Pardon my ignorance, but when will this not be the case?

Reply:
When the target is a number, string, object, etc.

Reply:
I see, the argument has already been parsed at this point into some Go data type beyond just a slice of bytes.

Reply:
> I see, the argument has already been parsed at this point into some Go data type beyond just a slice of bytes.

Sort of. This happens at the stage where we have string extracted from URL parameter, and are trying to figure out how to make JSON of them. We know we want a JSON `"string"` value, but not what goes inside.

If the corresponding parameter is `[]byte`, the contents of the string need to be encoded. JSON source must be UTF-8, so we can't inject the bits directly[1]. By convention, implementations use base64. On the other hand, if the receiver expects a plain string, it won't do any decoding at all, so wrapping the value in base64 will result in a string of base64 digits rather than the value the user typed.

So what we're doing here is taking advantage of an extra bit the method "compiler" stashed to tell us whether we should encode or not.

\